src/IO/GeoJSONReader.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Brick\Geo\IO;
+
+use Brick\Geo\CoordinateSystem;
+use Brick\Geo\Exception\GeometryException;
+use Brick\Geo\Exception\GeometryIOException;
+use Brick\Geo\Geometry;
+use Brick\Geo\GeometryCollection;
+use Brick\Geo\LineString;
+use Brick\Geo\MultiLineString;
+use Brick\Geo\MultiPoint;
+use Brick\Geo\MultiPolygon;
+use Brick\Geo\Point;
+use Brick\Geo\Polygon;
+
+/**
+ * Builds geometries out of GeoJSON Text strings.
+ */
+class GeoJSONReader
+{
+    /**
+     * @param string $geojson The GeoJSON to read.
+     *
+     * @return GeometryCollection|Geometry
+     *
+     * @throws GeometryException If the GeoJSON file is invalid.
+     */
+    public function read(string $geojson) : Geometry
+    {
+        $geojsonArray = json_decode($geojson, true);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new GeometryIOException(json_last_error_msg(), json_last_error());
+        }
+
+        if (! is_array($geojsonArray)) {
+            throw GeometryIOException::invalidGeoJSON('Unable to parse GeoJSON String.');
+        }
+
+        $geometry = $this->readGeoJSON($geojsonArray);
+
+        return $geometry;
+    }
+
+    /**
+     * @param array $geojson
+     *
+     * @return Geometry
+     *
+     * @throws GeometryIOException
+     * @throws \Brick\Geo\Exception\CoordinateSystemException
+     * @throws \Brick\Geo\Exception\InvalidGeometryException
+     * @throws \Brick\Geo\Exception\UnexpectedGeometryException
+     */
+    protected function readGeoJSON(array $geojson) : Geometry
+    {
+        if (! isset($geojson['type']) || ! is_string($geojson['type'])) {
+            throw GeometryIOException::invalidGeoJSON('Missing or Malformed "type" attribute.');
+        }
+
+        switch ($geojson['type']) {
+            case 'Feature':
+                return $this->readFeature($geojson);
+
+            case 'FeatureCollection':
+                // Verify 'FEATURES' exists
```

Comment:
Should be lowercase 'features'; actually this comment is not really necessary, the code speaks by itself.